### DISCUSSION TITLE
how to custom  http header in  gatsby

### DISCUSSION THREAD

**Discussion by menglingyu** [gatsbyjs/gatsby#30831] [Help]
how to custom  http header in  gatsby

**Comment by rodneylab** [comment]
Hi Meng Ling Yu,
I think this depends on where you are hosting you site.  If you use Netlify, you can the <a aria-label="Open gatsby-plugin-netlify docs" href="https://www.gatsbyjs.com/plugins/gatsby-plugin-netlify/">gatsby-plugin-netlify</a> plugin.  You can configure HTTP headers in the `gatsby-config.js` file once the plugin is installed.
Let me know if I can further help you.
Rodney

**Reply by menglingyu** [reply]
thanks  for your help , i want to host it  on my own server.  use nginx static resource server.

what plugins to use

**Reply by rodneylab** [reply]
Hi Meng Ling Yu,
In this case I would recommend that you set the headers yourself in the nginx config instead of using Gatsby plugins.  You can use the HeadersMore module (https://docs.nginx.com/nginx/admin-guide/dynamic-modules/headers-more/).

Depending on your experience, you might also find <a aria-label="Open Scott Helm's post on Hardening your HTTP response headers" href="https://scotthelme.co.uk/hardening-your-http-response-headers/">Scott Helme's post on Hardening HTTP response headers</a> useful.  He covers nginx in the post.

Hope that helps,
Rodney

**Reply by ucimo** [reply]
@rodneylab Hello. I wanted to change headers for data that's fetched during build on all routs that start with a standard language string, like "en/" for english. I need this for Shopify's translations. I tried using the plugin you suggested above, but it's not working. Any idea how can I achieve this?

**Reply by rodneylab** [reply]
Hi @obrovacm, how are you deploying your site? It might be possible to change the headers based on a route pattern match with Cloudflare Pages and some other services.

**Reply by ucimo** [reply]
I'm using Netlify, that is how I stumbled upon this discussion.

**Comment by rodneylab** [comment]
OK. Do you have to generate headers during build? You can add custom HTTP headers either in a netlify.toml file or by adding a _headers file to your build.  You could generate the _headers file yourself in a node script and output it to the static folder in your repo. Netlify will then serve the matching pages with the header you supply.

The [Netlify docs](https://docs.netlify.com/routing/headers/) mentions you can add * to match anything, would this fit your needs?

```
/example/en/*
  # headers for that path:
  X-Frame-Options: DENY
  SHOPIFY_HEADER: VALUE
```
